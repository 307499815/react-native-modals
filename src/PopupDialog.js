// flow

import React, {PropTypes, Component} from 'react';
import Dialog from './components/Dialog';
import DialogTitle from './components/DialogTitle';
import ScaleAnimation from './animations/ScaleAnimation';

const propTypes = {
	...Dialog.propTypes,
	...DialogTitle.propTypes,
};

const defaultProps = {
	animation: 'scale',
	animationDuration: 200,
	closeOnTouchOutside: true,
	dialogAnimation: new ScaleAnimation(),
};

class PopupDialog extends Component {
	static propTypes = propTypes;
	static defaultProps = defaultProps;
	
	openDialog(onOpened) {
		this.dialog.open(onOpened);
	}
	
	closeDialog(onClosed) {
		this.dialog.closed(onClosed);
	}
	
	render() {
		let title;
		
		if (this.props.title) {
			title = <DialogTitle {...this.props} />;
		}
		
		return (
			<Dialog
				ref={(dialog) => {
					this.dialog = dialog;
				}}
				{...this.props.children}
			>
				{title}
			</Dialog>
		);
	}
}

export default PopupDialog;
