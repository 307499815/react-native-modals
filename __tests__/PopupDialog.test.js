import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';

import PopupDialog, {
  DialogTitle,
  FadeAnimation,
  ScaleAnimation,
  SlideAnimation,
} from '../src';

it('should render with Default Animation correctly', () => {
  const wrapper = shallow((
    <PopupDialog
      dialogTitle={<DialogTitle title="Popup Dialog - Default Animation" />}
    />
  ));

  expect(wrapper.instance()).toMatchSnapshot();
  // show dialog
  wrapper.setProps({ show: true });
  expect(wrapper.instance()).toMatchSnapshot();
  // dismiss dialog
  wrapper.setProps({ show: false });
  expect(wrapper.instance()).toMatchSnapshot();
});

it('should render with FadeAnimation correctly', () => {
  const fadeAnimation = new FadeAnimation({
    toValue: 0,
    animationDuration: 200,
    useNativeDriver: true,
  });
  const wrapper = shallow((
    <PopupDialog
      dialogTitle={<DialogTitle title="Popup Dialog - FadeAnimation Animation" />}
      dialogAnimation={fadeAnimation}
    />
  ));

  expect(wrapper.instance()).toMatchSnapshot();
  // show dialog
  wrapper.setProps({ show: true });
  expect(wrapper.instance()).toMatchSnapshot();
  // dismiss dialog
  wrapper.setProps({ show: false });
  expect(wrapper.instance()).toMatchSnapshot();
});

it('should render with ScaleAnimation correctly', () => {
  const scaleAnimation = new ScaleAnimation({
    toValue: 0,
    useNativeDriver: true,
  });
  const wrapper = shallow((
    <PopupDialog
      dialogTitle={<DialogTitle title="Popup Dialog - ScaleAnimation Animation" />}
      dialogAnimation={scaleAnimation}
    />
  ));

  expect(wrapper.instance()).toMatchSnapshot();
  // show dialog
  wrapper.setProps({ show: true });
  expect(wrapper.instance()).toMatchSnapshot();
  // dismiss dialog
  wrapper.setProps({ show: false });
  expect(wrapper.instance()).toMatchSnapshot();
});

it('should render with SlideAnimation correctly', () => {
  const slideAnimation = new SlideAnimation({
    toValue: 0,
    slideFrom: 'bottom',
    useNativeDriver: true,
  });
  const wrapper = shallow((
    <PopupDialog
      dialogTitle={<DialogTitle title="Popup Dialog - SlideAnimation Animation" />}
      dialogAnimation={slideAnimation}
    />
  ));

  expect(wrapper.instance()).toMatchSnapshot();
  // show dialog
  wrapper.setProps({ show: true });
  expect(wrapper.instance()).toMatchSnapshot();
  // dismiss dialog
  wrapper.setProps({ show: false });
  expect(wrapper.instance()).toMatchSnapshot();
});
