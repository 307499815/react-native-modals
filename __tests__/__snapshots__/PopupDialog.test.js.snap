// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with Default Animation correctly 1`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - Default Animation"
      titleAlign="center"
    />,
  },
  "refs": Object {},
  "show": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - Default Animation"
            titleAlign="center"
          />
        }
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 150,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - Default Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with Default Animation correctly 2`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - Default Animation"
      titleAlign="center"
    />,
    "show": true,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - Default Animation"
            titleAlign="center"
          />
        }
        show={true}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 150,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={true}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - Default Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with Default Animation correctly 3`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - Default Animation"
      titleAlign="center"
    />,
    "show": false,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - Default Animation"
            titleAlign="center"
          />
        }
        show={false}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 150,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - Default Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with FadeAnimation correctly 1`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": FadeAnimation {
      "animate": 0,
      "animationDuration": 200,
      "animations": Object {
        "opacity": 0,
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - FadeAnimation Animation"
      titleAlign="center"
    />,
  },
  "refs": Object {},
  "show": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - FadeAnimation Animation"
            titleAlign="center"
          />
        }
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - FadeAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with FadeAnimation correctly 2`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": FadeAnimation {
      "animate": 0,
      "animationDuration": 200,
      "animations": Object {
        "opacity": 0,
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - FadeAnimation Animation"
      titleAlign="center"
    />,
    "show": true,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - FadeAnimation Animation"
            titleAlign="center"
          />
        }
        show={true}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={true}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - FadeAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with FadeAnimation correctly 3`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": FadeAnimation {
      "animate": 0,
      "animationDuration": 200,
      "animations": Object {
        "opacity": 0,
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - FadeAnimation Animation"
      titleAlign="center"
    />,
    "show": false,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - FadeAnimation Animation"
            titleAlign="center"
          />
        }
        show={false}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          FadeAnimation {
            "animate": 0,
            "animationDuration": 200,
            "animations": Object {
              "opacity": 0,
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - FadeAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with ScaleAnimation correctly 1`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": ScaleAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "scale": 0,
          },
          Object {
            "scale": 0,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - ScaleAnimation Animation"
      titleAlign="center"
    />,
  },
  "refs": Object {},
  "show": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - ScaleAnimation Animation"
            titleAlign="center"
          />
        }
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - ScaleAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with ScaleAnimation correctly 2`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": ScaleAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "scale": 0,
          },
          Object {
            "scale": 0,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - ScaleAnimation Animation"
      titleAlign="center"
    />,
    "show": true,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - ScaleAnimation Animation"
            titleAlign="center"
          />
        }
        show={true}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={true}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - ScaleAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with ScaleAnimation correctly 3`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": ScaleAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "scale": 0,
          },
          Object {
            "scale": 0,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - ScaleAnimation Animation"
      titleAlign="center"
    />,
    "show": false,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - ScaleAnimation Animation"
            titleAlign="center"
          />
        }
        show={false}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          ScaleAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - ScaleAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with SlideAnimation correctly 1`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": SlideAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "translateY": 800,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - SlideAnimation Animation"
      titleAlign="center"
    />,
  },
  "refs": Object {},
  "show": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - SlideAnimation Animation"
            titleAlign="center"
          />
        }
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - SlideAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with SlideAnimation correctly 2`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": SlideAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "translateY": 800,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - SlideAnimation Animation"
      titleAlign="center"
    />,
    "show": true,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - SlideAnimation Animation"
            titleAlign="center"
          />
        }
        show={true}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={true}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - SlideAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;

exports[`should render with SlideAnimation correctly 3`] = `
PopupDialog {
  "context": Object {},
  "dismiss": [Function],
  "props": Object {
    "dialogAnimation": SlideAnimation {
      "animate": 0,
      "animations": Object {
        "transform": Array [
          Object {
            "translateY": 800,
          },
        ],
      },
      "useNativeDriver": true,
    },
    "dialogTitle": <DialogTitle
      haveTitleBar={true}
      title="Popup Dialog - SlideAnimation Animation"
      titleAlign="center"
    />,
    "show": false,
  },
  "refs": Object {},
  "show": [Function],
  "state": Object {},
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <PopupDialog
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dialogTitle={
          <DialogTitle
            haveTitleBar={true}
            title="Popup Dialog - SlideAnimation Animation"
            titleAlign="center"
          />
        }
        show={false}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <Dialog
        animationDuration={150}
        containerStyle={null}
        dialogAnimation={
          SlideAnimation {
            "animate": 0,
            "animations": Object {
              "transform": Array [
                Object {
                  "translateY": 800,
                },
              ],
            },
            "useNativeDriver": true,
          }
        }
        dismissOnHardwareBackPress={true}
        dismissOnTouchOutside={true}
        haveOverlay={true}
        height={300}
        onDismissed={[Function]}
        onShown={[Function]}
        show={false}
        width={750}
      >
        <DialogTitle
          haveTitleBar={true}
          title="Popup Dialog - SlideAnimation Animation"
          titleAlign="center"
        />
      </Dialog>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
}
`;
